src/content/development/release-process/_index.en.md

```diff
@@ -36,19 +36,20 @@ Version numbers are required to be formatted following the schema norms where th
 
 ### Creating Releases
 
-The following sections are an ordered series of steps to create a Submariner relese.
+The following sections are an ordered series of steps to create a Submariner release.
 
-The release process is mostly automated and uses a YAML file created in the releases repository that describes the release.
+The release process is mostly automated and uses a YAML file created in the
+[releases repository](https://github.com/submariner-io/releases) that describes the release.
 This file is updated for each step in the release process.
 
 Once the changes for a step are reviewed and merged, a CI job will run to create the release(s) for the step and create the
 required pull requests in preparation for the next step to be reviewed and merged. Once all these pull requests have been
 merged, you can continue onto the next step.
 
-For most projects, after a release is created, another job will be initiated to build release artifacts and publish to Quay.
+For most projects, after a release is created, another job will be initiated to build release artifacts and publish it to Quay.
 This will take several minutes. You can monitor the progress from the project's main page. In the branches/tags pull-down
 above the file list heading, select the tag for the new version. A small yellow circle icon should be present to the right
-of the file list heading which indicates a job is in progress. You can click it to see details. There may be several checks
+of the file list heading which indicates a job is in progress. You can click on it to see details. There may be several checks
```

Comment:
The guidance I'm seeing is that it should remain as-is.

https://helpcenter.veeam.com/docs/styleguide/tw/click.html